lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,206 @@
+      return Dialog(
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: SingleChildScrollView(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                Center(
+                  child: Text('Settings',
+                      style: Theme.of(context).textTheme.headlineMedium),
+                ),
+                // Add a dropdown button for setting a new reference scale.
+                DropdownButton(
+                  alignment: Alignment.center,
+                  hint: const Text('Select a New Reference Scale'),
+                  // Set the selected scale
+                  value: _scale,
+                  icon: const Icon(
+                    Icons.arrow_drop_down,
+                    color: Colors.deepPurple,
+                  ),
+                  // Set the onChanged callback to update the selected scale.
+                  onChanged: (newScale) {
+                    setState(() => _scale = newScale!);
+                  },
+                  items: _referenceScaleList,
+                ),
+                const Text(
+                  'Apply Reference Scale to a Layer',
+                ),
+                // Add a list of checkboxes for selecting feature layers that will honor the reference scale.
+                Column(
+                  children: [
+                    // Create a checkbox for each feature layer.
+                    for (final layer in _allFeatureLayers)
+                      CheckboxListTile(
+                        value: _selectedFeatureLayers.contains(layer),
+                        onChanged: (value) {
+                          // Update the selected feature layers.
+                          setNewState(() {
+                            if (value ?? false) {
+                              _selectedFeatureLayers.add(layer);
+                            } else {
+                              _selectedFeatureLayers.remove(layer);
+                            }
+                          });
+                        },
+                        title: Text(layer),
+                      ),
+                  ],
+                ),
+                // Add a button to set the map scale to the reference scale and update the layers.
+                ElevatedButton(
+                  onPressed: () {
+                    setScaleAndLayers();
+                    Navigator.pop(context);
+                  },
+                  child: const Text('Set Map Scale to Reference Scale'),
+                ),
+              ],
+            ),
+          ),
+        ),
+      );
+    });
+  }
+
+  void setScaleAndLayers() {
+    // Set the map scale to the reference scale.
+    _map.referenceScale = _scale;
+    // Update the layers to honor the reference scale.
+    for (final layer in _selectedFeatureLayers) {
+      var matchingLayers = _map.operationalLayers
```

Comment:
Since `matchingLayers` will only be one thing, it should be `matchingLayer`. The plural kinda confused me.